### PR TITLE
[SPARK-41181][SQL] Migrate the map options errors onto error classes

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to migrate the map options errors onto error classes.

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@panbingkun Could you resolve conflicts, please.

**Comment by panbingkun** [comment]
> @panbingkun Could you resolve conflicts, please.

Done.

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -730,6 +730,23 @@
...
     ]
   },
+  "INVALID_OPTIONS" : {
+    "message" : [
+      "Invalid options:"
+    ],
+    "subClass" : {
+      "NON_STRING_TYPE" : {
+        "message" : [
+          "A type of keys and values in map() must be string, but got <map>."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:740]: nit: `map` -> `mapType`

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -730,6 +730,23 @@
...
+        ]
+      },
+      "NOT_MAP_FUNCTION" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:743]: Why not `NON_MAP_FUNCTION`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 3]
```diff
@@ -1013,13 +1013,13 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
 
   def keyValueInMapNotStringError(m: CreateMap): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1095",
-      messageParameters = Map("map" -> m.dataType.catalogString))
+      errorClass = "INVALID_OPTIONS.NON_STRING_TYPE",
+      messageParameters = Map("map" -> toSQLType(m.dataType)))
   }
 
   def nonMapFunctionNotAllowedError(): Throwable = {
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1020]: Be sure the names are consistent. not or non

File: core/src/main/resources/error/error-classes.json [hunk 4]
```diff
@@ -730,6 +730,23 @@
...
+        "message" : [
+          "Must use a map() function for options."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:745]:
```suggestion
          "Must use the `map()` function for options."
```

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**